baybe/searchspace/discrete.py

```diff
@@ -697,27 +728,34 @@ def _apply_polars_constraint_filter(
     return ldf
 
 
-def _apply_constraint_filter(
-    df: pd.DataFrame, constraints: Collection[DiscreteConstraint]
-):
-    """Remove discrete search space entries inplace based on constraints.
+def parameter_cartesian_prod_polars(parameters: Iterable[Parameter]) -> pl.LazyFrame:
+    """Create the Cartesian product of all parameter values.
+
+    Ignores continuous parameters.
 
     Args:
-        df: The data in experimental representation to be modified inplace.
-        constraints: List of discrete constraints.
+        parameters: List of parameter objects.
 
+    Returns:
+        A lazy dataframe containing all possible discrete parameter value combinations.
     """
-    # Reorder the constraints according to their execution order
-    constraints = sorted(
-        constraints,
-        key=lambda x: DISCRETE_CONSTRAINTS_FILTERING_ORDER.index(x.__class__),
-    )
+    discrete_parameters = [p for p in parameters if isinstance(p, DiscreteParameter)]
```

Comment:
Since we are in a discrete sub space, can it even happen that we have non-discrete parameters? Or is this check just to be sure?

Reply:
I didn't add this line, I only renamed the function  `_apply_constraint_filter` to ` _apply_pandas_constraint_filter` and it was part of `_apply_constraint_filter`.

Reply:
this check should stay here in analogy to what happens in the pandas version `parameter_cartesian_prod_to_df`
However the line should look more like `discrete_parameters = [p for p in parameters if p.is_discrete]`

Reply:
Changing this line causes mypy error since I'm using `p.values`

Reply:
you can ignore this error with a ignore statement because in this case mypy is wrong, this is also done in `parameter_cartesian_prod_to_df `

Reply:
@Scienfitz: I'm not sure if we've already had this conversation but: why do we actually introduce this property when the only real effect is that we lose the ability to typeguard?

Reply:
i think you yourself said sometime ago that we need to get rid of many `isinstance` calls that were there previously and I agree, the properties are the perfect way to do that, I dont think we should build our code to work aorund limitations of mypy

in this concrete case here nothing new has been introduced, the applications should just mimic what happens in the preexisting pandas constraint application function

Reply:
alright 👍🏼 in fact, perhaps we can solve the issue from a different angle in the future by
* upgrading mypy
* using some other checker such as basedmypy / pyright / etc....

But out of scope for this PR